js2-mode.el

```diff
@@ -7430,21 +7441,60 @@ When passed arguments of wrong type, does nothing."
           ;; to us before they are added as children of the assignment node.
           (js2-record-object-literal node qname (js2-node-pos node)))))))))
 
-(defun js2-compute-nested-prop-get (node)
+(defun js2-format-call-node-args (args)
+  "Format ARGS from JS function to string."
+  (mapconcat (lambda (arg)
+               (cond
+                ((js2-number-node-p arg)
+                 (format "%d" (js2-number-node-num-value arg)))
+                ((js2-string-node-p arg)
+                 (js2-node-string arg))
+                ((js2-function-node-p arg)
+                 "fn{..}")
+                (t
+                 "{}")))
+             args
+             ", "))
+
+(defun js2-call-node-output-for-compute-nesed-prop (node &optional extra-args)
+  "Format NODE and EXTRA-ARGS to string.  NODE corresponds to the JS function name and EXTRA-ARGS to parameters."
+  (list (if (and extra-args (js2-name-node-p node))
+            (format "%s(%s)"
+                    (js2-name-node-name node)
+                    (js2-format-call-node-args extra-args))
+          node)))
+
+(defun js2-compute-nested-prop-get (node &optional extra-args)
   "If NODE is of form foo.bar, foo['bar'], or any nested combination, return
 component nodes as a list.  Otherwise return nil.  Element-gets are treated
-as property-gets if the index expression is a string, or a positive integer."
+as property-gets if the index expression is a string, or a positive integer.
+If NODE is `js2-call-node', EXTRA-ARGS is NOT nil."
   (let (left right head)
     (cond
      ((or (js2-name-node-p node)
           (js2-this-or-super-node-p node))
-      (list node))
+      (js2-call-node-output-for-compute-nesed-prop node extra-args))
+     ;; For JS code `baz.setTimeout(function() { ... }, 1000);`:
+     ;;  `setTimeout(function() { ... }, 1000)` is the `js2-call-node'
+     ;;  `setTimeout` is the name of target of `js2-call-node'
+     ((js2-call-node-p node)        ; foo.bar
+      (js2-compute-nested-prop-get (js2-call-node-target node)
+                                   (js2-call-node-args node)))
      ;; foo.bar.baz is parenthesized as (foo.bar).baz => right operand is a leaf
      ((js2-prop-get-node-p node)        ; foo.bar
       (setq left (js2-prop-get-node-left node)
             right (js2-prop-get-node-right node))
-      (if (setq head (js2-compute-nested-prop-get left))
-          (nconc head (list right))))
+      ;; For JS code like `foo.state(...)`':
```

Comment:
`foo.state(...)` is a function call, not a function definition. Why would we show it in the index?